sdk/storage/azure-storage-file-share/tests/test_file.py

```diff
@@ -831,6 +829,39 @@ def test_set_file_properties(self, **kwargs):
         assert properties.creation_time is not None
         assert properties.permission_key is not None
 
+    @FileSharePreparer()
+    @recorded_by_proxy
+    def test_set_datetime_all_ms_precision(self, **kwargs):
+        storage_account_name = kwargs.pop("storage_account_name")
+        storage_account_key = kwargs.pop("storage_account_key")
+
+        self._setup(storage_account_name, storage_account_key)
+        file_client = self._create_file()
+
+        date_times = [
+            datetime(3005, 5, 11, 12, 24, 7),
+            datetime(3005, 5, 11, 12, 24, 7, 0),
+            datetime(3005, 5, 11, 12, 24, 7, 1),
+            datetime(3005, 5, 11, 12, 24, 7, 12),
+            datetime(3005, 5, 11, 12, 24, 7, 123),
+            datetime(3005, 5, 11, 12, 24, 7, 1234),
+            datetime(3005, 5, 11, 12, 24, 7, 12345),
+            datetime(3005, 5, 11, 12, 24, 7, 123456),
+            datetime(2023, 12, 8, tzinfo=timezone(-timedelta(hours=7))),
+            datetime(2023, 12, 8, tzinfo=timezone(-timedelta(hours=8))),
+        ]
+
+        # Act / Assert
+        content_settings = ContentSettings(
+            content_language='spanish',
+            content_disposition='inline')
```

Comment:
nit: Could probably leave these out to make the test a little simpler/smaller. Then you'd have to re-record though so up to you.

Reply:
Required positional argument!!!

Reply:
oof, ok then I guess.